index.bs

```diff
@@ -1598,286 +1600,161 @@ A URL for Blob and File reference</h2>
-a <a for=url>host</a> of the [=url/origin=] of the <a>Blob URL</a>,
-a <a for=url>path</a> with one entry comprised of a UUID [[RFC4122]]
-(see [[#ABNFUUID]]),
-and an <a for=url>object</a> of the associated {{Blob}} or {{File}}.
-A Blob URL may contain an optional <a for=url>fragment</a>.
-The Blob URL is serialized as a string according to the <a>Serialization of a Blob URL</a> algorithm.
-
-A <a for=url>fragment</a>, if used,
-has a distinct interpretation depending on the media type of the {{Blob}} or {{File}} resource in question
-(see [[#fragmentDiscussion]]).
-
-<pre>
-  blob = scheme ":" origin "/" UUID [fragIdentifier]
-
-  scheme = "blob"
-
-  ; scheme is always "blob"
-
-  ; origin is a string representation of the Blob URL's [=url/origin=].
-  ; UUID is as defined in <a biblio data-biblio-type=informative>\[RFC4122]</a> and <a href="#ABNFUUID"></a>
-  ; fragIdentifier is optional and as defined in <a biblio data-biblio-type=informative>\[RFC3986]</a> and <a href="#ABNFFragment"></a>
-</pre>
+Note: other specifications, such as [[MEDIA-SOURCE]] extend this scheme to also refer to other types of objects.
+
+<h3 id="url-intro">
+Introduction</h3>
+
+*This section is informative.*
+
+[=Blob URL|Blob (or object) URLs=] are URLs like
+<code>blob:http://example.com/550e8400-e29b-41d4-a716-446655440000</code>.
+This enables integration of {{Blob}}s and {{File}}s with other Web Platform APIs
+that are only designed to be used with URLs, such as the <{img}> element.
+[=Blob URLs=] can also be used to navigate to as well as to trigger downloads of
+locally generated data.
+
+For this purpose two static methods are exposed on the {{URL}} interface,
+{{URL/createObjectURL(blob)}} and {{URL/revokeObjectURL(url)}}.
+The first method creates a mapping from a URL to a Blob,
+and the second method revokes said mapping.
+As long as a mapping exist a Blob  can't be garbage collected,
+so some care must be taken to revoke a URL as soon as a reference is no longer needed.
+All URLs are revoked when the global that created the URL itself goes away.
+
+<h3 id="url-model">
+Model</h3>
+
+Each user agent must maintain a <dfn id="BlobURLStore" export>Blob URL Store</dfn>.
+A [=Blob URL Store=] is a [=map=]
+where [=map/keys=] are [=valid URL strings=]
+and [=map/values=] are [=Blob URL Entries=].
+
+A <dfn>Blob URL Entry</dfn> consists of
+an <dfn for="Blob URL Entry">object</dfn> (typically a {{Blob}},
+but other specs can extend this to refer to other types of objects),
+and an <dfn for="Blob URL Entry">environment</dfn> (an [=environment settings object=]).
+
+[=map/Keys=] in the [=Blob URL Store=] (also known as <dfn lt="Blob URL" export>Blob URLs</dfn>)
+are [=valid URL strings=] that when [=URL parser|parsed=]
+result in a [=/URL=] with a [=url/scheme=] equal to <code>"blob"</code>,
+a [=url/host=] equal to the [=environment settings object/origin=]
+of the [=environment settings object=] in which the URL was create,
```

Comment:
s/was create/was created

Is this the origin of the corresponding blob URL entry's environment? If so maybe I would make this correspondence explicit somewhere.

If it does *not* correspond, then an example of when that's the case would also be useful.